lib/Wikia/src/Consul/Client.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * A helper class that uses sensiolabs/consul-php-sdk for getting the list of service nodes
+ *
+ * @see PLATFORM-1489
+ * @see https://www.consul.io/docs/agent/http/health.html#health_service
+ */
+
+namespace Wikia\Consul;
+
+use SensioLabs\Consul\ServiceFactory;
+use Wikia\Logger\WikiaLogger;
+
+class Client {
+
+	use \Wikia\Logger\Loggable;
+
+	protected $logger;
+
+	/* @var \SensioLabs\Consul\Services\Health $api */
+	protected $api;
+
+	function __construct() {
+		$this->logger = WikiaLogger::instance();
+
+		$consulService = new ServiceFactory( [], $this->logger );
+		$this->api = $consulService->get( 'health' );
+	}
+
+	/**
+	 * Returns IP addresses of given service healthy nodes
+	 *
+	 * $catalog->getNodes( 'db-a', 'slave' )
+	 *
+	 * $ curl "http://127.0.0.1:8500/v1/health/service/db-g?tag=slave&passing"
+	 *
+	 * @param string $service
+	 * @param string $tag
+	 * @return array list of IP addresses
+	 */
+	function getNodes( $service, $tag ) {
+		$resp = $this->api->service( $service, [ 'tag' => $tag, 'passing' => true ] )->json();
+
+		$nodes = array_map(
+			function( $item ) {
+				return $item[ 'Node' ][ 'Address' ];
+			},
+			$resp
+		);
+
+		wfDebug( __METHOD__ . sprintf( ": got nodes for '%s' service ('%s' tag): %s\n", $service, $tag, join(', ', $nodes) ) );
+		return $nodes;
+	}
+
+	/**
+	 * Helper method for getting database nodes hidden behind consul
+	 *
+	 * $catalog->getDatabaseNodes( 'slave.db-smw.service.consul' )
+	 *
+	 * @param $db
+	 * @return array list of IP addresses
+	 */
+	function getDatabaseNodes( $db ) {
+		list( $tag, $service ) = explode( '.', $db );
+		return $this->getNodes( $service, $tag );
+	}
+
+	/**
+	 * Example: Wikia\Consul\Catalog::isConsulAddress( 'slave.db-smw.service.consul' )
+	 *
+	 * @param $address
+	 * @preturn bool true if the given address is a consul one
+	 */
+	static function isConsulAddress( $address ) {
+		return endsWith( $address, '.service.consul' );
+	}
+
+	/**
+	 * Make it easier to "grep" through the logs
+	 *
+	 * @return array
+	 */
+	protected function getLoggerContext() {
+		return [
+			'class' => __CLASS__
+		];
+	}
+}
```

Comment:
Yeah, I agree that this is probably better than interacting directly with the API. But we should consolidate on one. If we use this, it should provide a module to be inject-able within other `lib/Wikia` stuff